### PR TITLE
Dockerize Build Harness & Add Add Helm Chart Targets

### PR DESCRIPTION
**Pull request by goruha**
## What
* Dockerize build-harness
* Added git module to analyze git history
* Added git doc module
* Added helm chart ops
* Small refactoring of helm module
* Added semver module

## Why
* To use with codefresh
* Allow to do different ops in other modules depend of - are we on tag or branch
* Allow to inject git data into gomplate templates
* Allow to build and publish single chart to helm repo (based on chartmuseum)
* Because helm module now operates on helm repo and helm chart it need some cosmetic changes
* Made POC of versioning convention for charts and docker images

## Demo
* `make semver:show` generates list of potential version based on git history
```
0.4.0-sha.90a3324
0.4.0-sha.90a3324baeeae9b345fb51f505f00a97b54e50ce
0.4.0-master
0.4.0-master.90a3324
0.4.0-master.90a3324baeeae9b345fb51f505f00a97b54e50ce
```

### CODE REVIEW

File: modules/helm/Makefile.repo [hunk 1]
```diff
@@ -31,79 +28,96 @@ helm\:repo\:info:
...
-helm\:repo\:add:
-	@$(HELM) repo add kubernetes-charts http://storage.googleapis.com/kubernetes-charts
-	@$(HELM) repo add opsgoodness http://charts.opsgoodness.com
-	@$(HELM) repo add kubernetes-charts-incubator https://kubernetes-charts-incubator.storage.googleapis.com
+helm\:repo\:add-all:
+	@REPO_NAME=kubernetes-charts \
+		REPO_ENDPOINT=http://storage.googleapis.com/kubernetes-charts \
+		make -s --no-print-directory helm:repo:add
+	@REPO_NAME=opsgoodness \
+		REPO_ENDPOINT=http://charts.opsgoodness.com \
```
Comment (osterman) [inline modules/helm/Makefile.repo]: Does this still work? Is it still relevant?

File: modules/semver/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+SEMVERSIONS :=
```
Comment (osterman) [inline modules/semver/Makefile:2]: Please annotate this `Makefile` to make it more clear what it's doing.

File: modules/helm/Makefile.chart [hunk 3]
```diff
@@ -0,0 +1,67 @@
+HELM_STARTERS ?= $(HELM_HOME)/starters
+CLOUDPOSSE_HELM_STARTER_REPO ?= https://github.com/cloudposse/helm-chart-scaffolding.git
+CLOUDPOSSE_HELM_STARTER_REF ?= master
+CLOUDPOSSE_HELM_STARTER_NAME ?= cloudposse
+
+CHART_TPL ?= chart
+
+## Fetch starter
+helm\:chart\:starter\:fetch:
```
Comment (osterman) [inline modules/helm/Makefile.chart]: Starter should be renamed `scaffolding`

File: modules/helm/Makefile.repo (hunk 1 above)
Comment (goruha) [reply modules/helm/Makefile.repo]: Yes

File: modules/git/Makefile [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+endif
+
+
+ifeq ($(shell $(GIT) describe --exact-match --tags 2>/dev/null),)
+	export GIT_IS_TAG := 0
+else
+	export GIT_IS_TAG := 1
+endif
+
+define git_vars_export
```
Comment (osterman) [inline modules/git/Makefile]: Clever hack, but let's just explicitly `echo` them below.

File: modules/git/docs/module.sh [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
+  latest_tag: $(LATEST_TAG)
+  is_tag: $(GIT_IS_TAG)
+  is_branch: $(GIT_IS_BRANCH)
 EOF
 }
 
-function git_data-docs-cleanup-data {
-  rm -f $GIT_DATA_FILE
+function git-docs-cleanup-data {
+  rm -f $GIT_FILE
```
Comment (osterman) [inline modules/git/docs/module.sh]: Use quotes around variables

File: modules/git/docs/module.sh [hunk 6]
```diff
@@ -1,18 +1,23 @@
...
+GIT_FILE=$TMP/git.yml
 
-DATASOURCES[git]=file://$BUILD_HARNESS_PATH/modules/git/docs/templates/git.yml
-DATASOURCES[git_data]=file://$GIT_DATA_FILE
+DATASOURCES[git]=file://$GIT_FILE
 
-function git_data-docs-prepare-data {
-cat << EOF > $GIT_DATA_FILE
+function git-docs-prepare-data {
+cat << EOF > $GIT_FILE
```
Comment (osterman) [inline modules/git/docs/module.sh]: Use quotes

File: modules/git/docs/module.sh [hunk 7]
```diff
@@ -1,18 +1,23 @@
 #!/usr/bin/env bash
 
-GIT_DATA_FILE=$TMP/git_data.yml
+GIT_FILE=$TMP/git.yml
```
Comment (osterman) [inline modules/git/docs/module.sh]: use quotes

File: modules/helm/Makefile [hunk 8]
```diff
@@ -10,7 +14,7 @@ helm\:install:
 	@chmod +x $(HELM_PLATFORM)/helm
 	@mv $(HELM_PLATFORM)/helm /usr/local/bin/
 	@helm init --client-only
-	@chmod -R 777 $(HOME)/.helm
+	@chmod -R 777 $(HELM_HOME)
```
Comment (osterman) [inline modules/helm/Makefile]: Use quotes

**Review by osterman — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**